### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by StevenWafeek**
in this project :
- Create a class Classroom 
- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
1. Create the has-many side (a classroom has many students).
2. Create the belongs-to side (a student belongs to a classroom).
- Create a class Book
- Create a class Rental
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ def correct_name
 end
 
 class Person < Nameable
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :rentals
```
Comment (elisha2kyakpo1) [inline person.rb:8]: - [ ] You did well to add a rental as a getter/setter, but we need to associate it using a method. Kindly add a method that will link a person and a rental, achieving the `has_many` rentals associations.

```Ruby
def add_rental(book, date)
    Rental.new(date, book, self)
end
```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = [] # Initialize rentals as an empty array
+  end
```
Comment (elisha2kyakpo1) [inline book.rb:8]: - [ ] Kindly add an association between a book and a person renting a book. A add rental method should be created to account for a book, the person renting, and the date it has/was rented. [Here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) is a link to the reason if you want to refresh.

File: student.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 require_relative 'person'
+require_relative 'classroom'
 
 class Student < Person
   attr_accessor :classroom
```
Comment (elisha2kyakpo1) [inline student.rb:5]: - [ ] Kindly add a method that will link a student to a classroom. A student belongs to a classroom.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @StevenWafeek, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @StevenWafeek      , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  . 
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] No linter errors
- [x] Database association correctly setup 💯


 

 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**